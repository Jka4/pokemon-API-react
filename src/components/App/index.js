import React from 'react';
import App from './view/index';

export default function () {
	return <App />;
}
