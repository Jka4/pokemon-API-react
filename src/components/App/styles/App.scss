@import-normalize;

body,
html {
  // width: 100%;
  height: auto;
}

body {
  background-color: rgba(255, 255, 0, 0.603);
  overflow-y: scroll;

}

* {
  font-family: "VT323", monospace;
}

.getButtonLine {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

#getPokemon {
  margin: 0 auto;
  margin-top: 10px;
  margin-right: auto;
  margin-left: auto;
  outline-style: none;
  border: 3px solid black;
  border-radius: 7px;
  background: red;
  cursor: pointer;
  padding: 3px;
  color: white;
  font-size: 28px;
}

#getPokemon:hover {
  box-shadow: 5px 3px 11px 1px #000000;
}

#getPokemon:active {
  bottom: -4px;
  box-shadow: 0 1px 0 #00823f;
}

.playButton:hover button {
  box-shadow: 5px 3px 11px 1px #000000;
}

.playButton:active button {
  bottom: -4px;
  box-shadow: 0 1px 0 #00823f;
}

.main {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  margin: 0 auto;
  width: 100%;
  max-width: 900px;
  height: auto;
}

ul {
  display: flex;
  flex-direction: row;
  margin-top: 20px;
}

li {
  list-style: none;
  text-transform: uppercase;
}

.itemList {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 90%;
  max-width: 900px;
}

.cardsContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding-bottom: 100px;
  width: 100%;
  height: auto;
}

.pokemonCard {
  border: 2px solid black;
  border-radius: 5px;
  background-color: rgba(250, 235, 215, 0.74);
  cursor: pointer;
  width: 100%;
  height: 100%;
  text-transform: uppercase;
  font-size: 22px;

  img,
  .imageWrapper {
    min-width: 100px;
    margin-left: auto;
    margin-right: auto;
    max-height: 100px;
  }

  svg {
    margin-left: auto;
    margin-right: auto;
  }

  div {
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

.pokemonCard_Outher {
  margin-top: 10px;
  width: 100%;
  height: 100px;
  text-decoration: none;
  color: black;
}

.pokemonCard img {
  height: 100%;
}

.pokemonCard:hover {
  transform: scale(1.02);
}

.pokemonCard:hover img {
  transform: scale(1.06);
}

.topLine {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 60px;
}

.playButton {
  display: flex;
  position: fixed;
  top: 10px;
  left: 10px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  outline-style: none;
  border: 3px solid black;
  border-radius: 7px;
  background: red;
  cursor: pointer;
  width: 90px;
  height: 40px;
  text-align: center;
  color: white;
  font-size: 29px;
  z-index: 2;
}

.search {
  display: flex;
  position: absolute;
  top: 10px;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  justify-self: center;
  width: auto;
  height: auto;
}

.search input {
  justify-self: center;
  z-index: 10;
  outline: none;
  border: 1px solid black;
  border-radius: 5px;
  padding-left: 20px;
  width: 30rem;
  max-width: 450px;
  height: 2.4rem;
  font-size: 1rem;
}

#searchInput::placeholder {
  font-size: 1.2rem;
}

#searchInput:focus {
  margin-right: 1px;
  margin-left: 1px;
  border: 1px transparent solid;
  padding-top: 2px;
  padding-bottom: 2px;
}

.searchResults {
  display: flex;
  position: relative;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  z-index: 10;
  margin-top: 5px;
  box-shadow: 0 1.2px 2.1px rgba(0, 0, 0, 0.07),
    0 2.9px 5.1px rgba(0, 0, 0, 0.101), 0 5.4px 9.6px rgba(0, 0, 0, 0.125),
    0 9.6px 17.2px rgba(0, 0, 0, 0.149), 0 18px 32.2px rgba(0, 0, 0, 0.18),
    0 43px 77px rgba(0, 0, 0, 0.25);
  width: 120%;
  height: auto;
}

.searchList {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  margin: 0;
  background: white;
  padding: 0;
  width: 100%;
}

.searchItem {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  margin: 0;
  border-bottom: 1px grey solid;
  padding: 0;
  width: 100%;
  height: 3rem;
  text-decoration: none;
  font-size: 1.1rem;
}

.searchItem_outer {
  width: 100%;
}

.item_name {
  width: 35%;
}

.item_id {
  width: 13%;
  text-align: left;
}

.item_weight {
  position: relative;
  right: 0;
  width: 22%;
  text-align: left;
}

.searchItem img {
  justify-self: flex-end;
  min-width: 45px;
}

.searchItem:hover {
  transform: scale(1.02);
  background: rgb(238, 238, 238);
  cursor: pointer;
}

.searchItem:hover img {
  transform: scale(1.17);
}

.searchItem:last-child {
  border-bottom: 1px transparent solid;
}

.searchItem__image {
  width: auto;
  height: 100%;
}

.Tilt,
.Tilt-inner {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.Tilt-inner {
  min-height: 212px;
}

.Tilt-inner svg {
  margin-top: 50px;
  margin-bottom: 40px;
}

.detailedPage {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  height: auto;
}

.detailedPage .name {
  width: 100%;
  height: auto;
  text-align: center;
  font-size: 4rem;
}

.detailedPage .name::first-letter {
  text-transform: uppercase;
}

.imagesLine {
  margin-top: 2rem;
  width: 100%;
  height: auto;
  max-width: 600px;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: 1fr;
  grid-column-gap: 20px;
  grid-row-gap: 20px;

  .block {
    margin-left: auto;
    margin-right: auto;
  }
}


.imagesLineSmall {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  max-width: 600px;
  margin-top: 30px;
}

.imagesLineSmall .block {
  margin-left: 20px;
  margin-right: 20px;
}

.imagesLine .block,
.imagesLineSmall .block {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;
  border: 1px solid black;
  border-radius: 0.7rem;
  background: #6cff79;
  cursor: default;
  padding: 1rem;
  padding-top: 5px;
  width: 10vw;
  max-width: 5rem;
  min-height: 6rem;
  max-height: 115px;
}

.imagesLine .block:hover,
.imagesLineSmall .block {
  transform: scale(1.1);
  box-shadow: 0 1.2px 2.1px rgba(0, 0, 0, 0.07),
    0 2.9px 5.1px rgba(0, 0, 0, 0.101), 0 5.4px 9.6px rgba(0, 0, 0, 0.125),
    0 9.6px 17.2px rgba(0, 0, 0, 0.149), 0 18px 32.2px rgba(0, 0, 0, 0.18),
    0 43px 77px rgba(0, 0, 0, 0.25);
}

.imagesLine .block img,
.imagesLineSmall .block img {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 90%;
  height: auto;
}

.block .title {
  overflow-x: auto;
  font-size: 1rem;
}

.skills {
  display: flex;
  flex-direction: row;
  align-items: flex-start;
  justify-content: center;
  margin-top: 2rem;
  width: 100%;
  height: auto;
}

.stats {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  margin-right: 3rem;
  width: 40vw;
}

.abilities {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin-left: -1rem;
  width: 40vw;
}

.statsUL {
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  justify-content: flex-start;
  margin-left: 0;
  border: 1px solid black;
  border-radius: 0.7rem;
  background: #53aeff;
  padding: 25px;
  padding-left: 15px;
}

.abilitiesUL {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;
  margin-left: 0;
  border: 1px solid black;
  border-radius: 0.7rem;
  background: #53aeff;
  padding: 25px;
  padding-left: 15px;
}

.stats .statsLI {
  margin-top: 1rem;
  margin-left: 0;
  text-align: right;
}

.abilitiesLI {
  margin-top: 1rem;
  margin-left: 0;
  text-align: left;
}

.skills_title {
  align-self: center;
  justify-self: center;
  font-size: 2rem;
}

#clear,
.backToMainPage {
  margin-right: auto;
  margin-left: 10px;
  outline-style: none;
  border: 3px solid black;
  border-radius: 7px;
  background: red;
  cursor: pointer;
  padding: 3px;
  color: white;
  font-size: 28px;
  min-width: 80px;
  text-align: center;
  z-index: 2;
}

a {
  text-decoration: none;
}

@media only screen and (min-device-width: 320px) and (max-device-width: 480px) {
  .topLine {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;
    width: 100%;
    height: 8vh;
  }

  .player {
    display: flex;
    position: absolute;
    top: 15px;
    left: 25px;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    z-index: 1000;
    width: auto;
    height: 40px;
  }

  .playButton {
    position: absolute;
    top: 0;
    left: 0;
    background: red;
    z-index: 123;
  }

  .playButton button {
    margin: 0;
  }

  .search {
    top: 15px;
    right: 25px;
    width: 60%;
  }

  .search input {
    margin-right: -3rem;
    padding-left: 10px;
    width: 80%;
    font-size: 1.4rem;
  }

  #searchInput::placeholder {
    font-size: 1.4rem;
  }

  .searchResults {
    position: absolute;
    top: 45px;
    right: -0.6rem;
    width: 92.5vw;
  }

  .getButtonLine {
    margin: 0;
  }

  .getButtonLine img {
    width: 85%;
  }

  .getButtonLine .Button {
    margin-top: 30px;
  }

  .main {
    padding-bottom: 50px;
    width: 100%;
    max-width: auto;
  }

  .main h2 {
    width: 100%;
  }

  .main ul {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;
    margin: 0;
    margin-top: 10px;
    padding: 0;
    width: auto;
  }

  .main li {
    margin: 0;
    padding: 0;
    width: auto;
  }

  .detailedPage {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: flex-start;
    margin-top: 10px;
    height: auto;
  }

  .imagesLine .block,
  .imagesLineSmall .block {
    margin-right: 10px;
    margin-bottom: 2rem;
    margin-left: 10px;
    width: 20vw;
  }

  .imagesLine .block .title,
  .imagesLineSmall .block .title {
    overflow: hidden;
    color: red;
  }

  .imagesLine .block img,
  .imagesLineSmall .block img {
    width: auto;
  }
}

.titles {
  width: 90%;
  max-width: 900px;

  .titlesList {
    display: grid;
    grid-template-columns: repeat(4, 1fr);
    grid-template-rows: 1fr;
    grid-column-gap: 20px;
    grid-row-gap: 0px;
    padding-left: 0;

    li {
      text-align: center;
    }
  }
}

.pokemonCard {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: 1fr;
  grid-column-gap: 20px;
  grid-row-gap: 0px;
  padding-left: 0;

  img {
    margin-left: auto;
    margin-right: auto;
  }

  span {
    text-align: center;
    margin-top: auto;
    margin-bottom: auto;
  }
}

.bigImage {
  img {
    height: 100%;
    max-height: 400px;
  }
}

.backToMainPage {
  position: fixed;
  top: 60px;
  left: 0px;
}

.searchItem {
  color: black;
}


.err {
  color: #ff3527;
}