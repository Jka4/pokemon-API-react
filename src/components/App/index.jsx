/* eslint-disable import/no-anonymous-default-export */
import React from "react";
import App from "@App";

export default function () {
  return <App />;
}
