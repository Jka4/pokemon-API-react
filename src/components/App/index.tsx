/* eslint-disable import/no-anonymous-default-export */
import React from 'react';
import App from './view/App';

export default () => {
  return <App />;
};
