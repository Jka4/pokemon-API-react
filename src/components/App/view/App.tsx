import React, { useEffect, useState, lazy, Suspense } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

import '../styles/App.scss';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';

const HeaderLine = lazy(() => import('components/HeaderLine/HeaderLine'));
const MainContainer = lazy(() => import('components/MainContainer/MainContainer'));
const DetailedPage = lazy(() => import('components/DetailedPage/DetailedPage'));
const PokemonPageAll = lazy(() => import('components/PokemonPageAll/index'));
const NonSupportPlaceholder = lazy(() => import('components/NonSupportPlaceholder/NonSupportPlaceholder'));
const ErrorPage = lazy(() => import('components/ErrorPage/ErrorPage'));

const App: React.FC = () => {
  const [supportScreenSize, setSupportScreenSize] = useState<boolean>(true);

  const getScreenWidth = () => {
    let res: boolean = document.body.clientWidth <= 319;
    setSupportScreenSize(res);
  };

  useEffect(() => {
    window.addEventListener('resize', getScreenWidth);
    getScreenWidth();
  });

  return (
    <Suspense fallback={<div className="fallback">Loading...</div>}>
      {!supportScreenSize ? (
        <div className="App">
          <Route exact component={HeaderLine} />

          <Switch>
            <Route exact path="/" component={MainContainer} />
            <Route path="/detailedPage/pokemon/" component={DetailedPage} />
            <Route path="/allPokemon" component={PokemonPageAll} />
            <Route path="/404" component={ErrorPage} />

            <Redirect from="*" to="/404" />
          </Switch>
        </div>
      ) : (
        <NonSupportPlaceholder />
      )}
    </Suspense>
  );
};

export default App;
