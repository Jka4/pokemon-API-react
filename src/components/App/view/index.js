import React from "react";
import { Route, BrowserRouter as Router, Switch } from "react-router-dom";

import "../styles/App.scss";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css"

import MainContainer from "@MainContainer";
import DetailedPage from "@DetailedPage";
import PokemonsPageAll from '@PokemonsPageAll';

const App = () => {
  return (
    <Router basename='/pokemon-API-react'>
      <div className='App'>
        <Switch>
          <Route exact path='/' component={MainContainer} />
          <Route path='/detailedPage/pokemon/' component={DetailedPage} />
          <Route path='/allPokemons' component={PokemonsPageAll} />
          <Route path="/" component={() => <div>page not found</div>} />
        </Switch>
      </div>
    </Router>
  );
}

export default App;
