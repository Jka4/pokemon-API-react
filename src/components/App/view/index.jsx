import React, { useEffect, useState } from "react";
import { Route, Switch, Redirect } from "react-router-dom";

import "../styles/App.scss";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";

import MainContainer from "@MainContainer";
import DetailedPage from "@DetailedPage";
import PokemonsPageAll from "@PokemonsPageAll";
import HeaderLine from "@HeaderLine";
import ErrorPage from "@ErrorPage";
import NonSupportPlaceholder from "@NonSupportPlaceholder";

const App = () => {
  const [supportScreenSize, setSupportScreenSize] = useState(null);

  const getScreenWidth = () => {
    setSupportScreenSize(document.body.clientWidth <= 319);
  };

  useEffect(() => {
    window.addEventListener("resize", getScreenWidth);
    getScreenWidth();
  });

  return (
    <>
      {!supportScreenSize ? (
        <div className="App">
          <Route render={(props) => <HeaderLine props={props} />} />

          <Switch>
            {/* <Route exact path="/" component={MainContainer} /> */}
            {/* <Route path="/detailedPage/pokemon/" component={DetailedPage} /> */}
            {/* <Route path="/allPokemons" component={PokemonsPageAll} /> */}
            {/* <Route path="/404" component={ErrorPage} /> */}

            {/* <Redirect from="*" to="/404" /> */}
          </Switch>
        </div>
      ) : (
        <NonSupportPlaceholder />
      )}
    </>
  );
};

export default App;
