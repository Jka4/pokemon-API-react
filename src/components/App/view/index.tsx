import React, { useEffect, useState, lazy, Suspense } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';

import '../styles/App.scss';
import 'react-loader-spinner/dist/loader/css/react-spinner-loader.css';

const HeaderLine = lazy(() => import('components/HeaderLine/HeaderLine'));
const MainContainer = lazy(() => import('components/mainContainer/mainContainer'));
const DetailedPage = lazy(() => import('components/DetailedPage/DetailedPage'));
const PokemonsPageAll = lazy(() => import('components/PokemonsPageAll/index'));
const ErrorPage = lazy(() => import('components/ErrorPage/ErrorPage'));
const NonSupportPlaceholder = lazy(() => import('components/nonSupportPlaceholder/nonSupportPlaceholder'));

const App: React.FC = () => {
    const [supportScreenSize, setSupportScreenSize] = useState<boolean>(true);

    const getScreenWidth = () => {
        let res: boolean = document.body.clientWidth <= 319;
        setSupportScreenSize(res);
    };

    useEffect(() => {
        window.addEventListener('resize', getScreenWidth);
        getScreenWidth();
    });

    return (
        <Suspense fallback={<div className="fallback">Loading...</div>}>
            {!supportScreenSize ? (
                <div className="App">
                    <Route exact component={HeaderLine} />

                    <Switch>
                        <Route exact path="/" component={MainContainer} />
                        <Route path="/detailedPage/pokemon/" component={DetailedPage} />
                        <Route path="/allPokemons" component={PokemonsPageAll} />
                        <Route path="/404" component={ErrorPage} />

                        <Redirect from="*" to="/404" />
                    </Switch>
                </div>
            ) : (<NonSupportPlaceholder />)}
        </Suspense>
    );
};

export default App;
