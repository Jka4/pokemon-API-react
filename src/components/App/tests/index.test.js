import React from 'react';
import App from '@App';
import renderer from 'react-test-renderer';
import { BrowserRouter } from "react-router-dom";

it('App renders correctly', async () => {
	const tree = renderer
		.create(<BrowserRouter><App /></BrowserRouter>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
