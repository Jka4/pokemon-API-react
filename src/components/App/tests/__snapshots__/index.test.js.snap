// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App renders correctly 1`] = `
<div
  className="fallback"
>
  Loading...
</div>
`;
